integration_tests/features/steps/fal_steps.py

```diff
@@ -219,43 +219,67 @@ def _verify_node_order(context):
         profiles_dir=_set_profiles_dir(context), project_dir=context.base_dir
     )
     node_graph = NodeGraph.from_fal_dbt(fal_dbt)
-    ordered_nodes = list(map(_as_name, _get_all_models(context)))
+
+    all_nodes = _get_dated_dbt_models(context) + _get_dated_fal_artifacts(
+        context, FAL_SCRIPT
+    )
+    # We need to normalize the suffix for Python models.
+    all_nodes += [
+        (_as_name(name), date)
+        for name, date in _get_dated_fal_artifacts(context, FAL_MODEL)
+    ]
+    ordered_nodes = _unpack_dated_result(all_nodes)
 
     graph = node_graph.graph
-    node_to_ancestors, node_to_post_hook = {}, {}
+    ancestors, post_hooks, pre_hooks = {}, {}, {}
     for node, data in graph.nodes(data=True):
         name = _as_name(node)
-        node_to_ancestors[name] = [
+        ancestors[name] = [
             _as_name(ancestor)
             for ancestor in nx.ancestors(graph, node)
             if _as_name(ancestor) in ordered_nodes
         ]
-        node_to_post_hook[name] = [
-            _script_filename(post_hook)
-            for post_hook in data.get("post_hook", [])
-            if _script_filename(post_hook) in ordered_nodes
-        ]
-
+        for container, hook_type in [
+            (pre_hooks, "pre_hook"),
+            (post_hooks, "post_hook"),
+        ]:
+            container[name] = [
```

Comment:
Wait, aren't you overwriting the value here?

Reply:
not overwriting, but setting. We only have 1 write against `container[name]`.